doobie/src/main/scala/tofu/doobie/transactor/Txr.scala

```diff
@@ -42,11 +42,16 @@ trait Txr[F[_]] {
 }
 
 object Txr {
-  type Aux[F[_], DB0[_]]     = Txr[F] { type DB[x] = DB0[x] }
+  type Aux[F[_], DB0[_]]     = Txr[F] { type DB[x] >: DB0[x] <: DB0[x] }
```

Comment:
Everything compiles fine with this change (including the example). The only issue with the previous encoding was related to `distage` runtime. I couldn't write a failing test with `izumi-reflect` only and I don't want to involve the whole `distage` lib into tests. I checked (and already knew for a long time) that this fix worked out.